### DISCUSSION TITLE
GPIOZ Missing in stm32mp157.dtsi

### DISCUSSION THREAD

**Discussion by georgenyarangi-eaton** [zephyrproject-rtos/zephyr#48331] [Q&A]
It looks like gpioz is missing from the pinctrl in [stm32mpl157.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/2495a06f1a0424b0d3df61d1e98a4dd22a311ca3/dts/arm/st/mp1/stm32mp157.dtsi#L53)

When I try add it in an overlay file as such:
```&pinctrl{
    gpioz: gpio@54004000 {
        compatible = "st,stm32-gpio";
        reg = <0x54004000 0x400>;
        gpio-controller;
        #gpio-cells = <2>;
        clocks = <&rcc STM32_CLOCK_BUS_AHB5 0x00000001>;
        label = "GPIOZ";
        status = "okay";
    };
};
```
and reference it in the root node as such:
```    mmplus_io {
        compatible = "mmplus-io";
        reset_line0: output0 {
            gpios = <&gpioe 15 GPIO_ACTIVE_LOW>;
            label = "CODEC_RST";
        };
        status_led0: led0 {
            gpios = <&gpioz 7 GPIO_ACTIVE_LOW>;
            label = "G_LED_N";
        };
    };

    aliases {
        codec-reset-line = &reset_line0;
        green-status-led = &status_led0;
    };
```
I get an undefined reference linker error when trying to use the `green-status-led` alias in my application:
`c:/arm-embedded/gcc-arm-none-eabi-10.3-2021.07/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld.exe: app/libapp.a(main.cpp.obj):(.rodata._ZL14green_led_spec+0x0): undefined reference to `__device_dts_ord_19'`

Note that the `codec-reset-line` works just fine. Also, I do have a yaml file for `mmplus_io`.

**Comment by erwango** [comment]
Unfortunately STM32 GPIO driver would require some changes to support a GPIO Z port.
See https://github.com/zephyrproject-rtos/zephyr/blob/3b5590a18790ea8f9159eadbc38f8a514b342543/drivers/gpio/gpio_stm32.c#L782-L789


Few other changes would be required to [dt-bindings/pinctrl/stm32-pinctrl-common.h](https://github.com/zephyrproject-rtos/zephyr/blob/3b5590a18790ea8f9159eadbc38f8a514b342543/include/zephyr/dt-bindings/pinctrl/stm32-pinctrl-common.h)

**Reply by georgenyarangi-eaton** [reply]
These changes seem to do the trick. Thank you!

**Reply by erwango** [reply]
Nice, don't hesitate to contribute if you're satisfied with these changes.
